### PR TITLE
Replace deprecated 5.x packages to 6.x

### PR DESCRIPTION
**Pull request by kshyju**
Upgrading deprecated 5.X version of packages to 6X. Also updated the test projects from net5 to net6.

Bumped patch version of our packages for this change since no public API surface/behavior exposed by us is changing with this. For AI project, bumped preview version.

The below top level packages are deprecated.

### In [src](https://github.com/Azure/azure-functions-dotnet-worker/tree/main/src)

- [Microsoft.Extensions.Hosting 5.0.0](https://www.nuget.org/packages/Microsoft.Extensions.Hosting/5.0.0)
- [Microsoft.Extensions.Hosting.Abstractions 5.0.0](https://www.nuget.org/packages/Microsoft.Extensions.Hosting.Abstractions/5.0.0)
- [Microsoft.Extensions.Options 5.0.0](https://www.nuget.org/packages/Microsoft.Extensions.Options/5.0.0)

------------------------------------
### In the [tests](https://github.com/Azure/azure-functions-dotnet-worker/tree/main/test)

- [Microsoft.Extensions.Configuration.Abstractions 5.0.0](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Abstractions/5.0.0)
- [Microsoft.Extensions.Configuration.EnvironmentVariables - 5.0.0](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.EnvironmentVariables/5.0.0)
- [Microsoft.Extensions.Configuration.Json - 5.0.0](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Json/5.0.0)
- [Microsoft.Extensions.Logging.Abstractions  - 5.0.0](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Abstractions/5.0.0)
- [Microsoft.Extensions.Logging - 5.0.0](https://www.nuget.org/packages/Microsoft.Extensions.Logging/5.0.0)

Also updated the below packages because they are transitive dependency of the above packages

[System.Text.Json - 5.0.2](https://www.nuget.org/packages/System.Text.Json/5.0.2)
[System.Collections.Immutable ](https://www.nuget.org/packages/System.Collections.Immutable/)

### CODE REVIEW

File: src/DotNetWorker/DotNetWorker.csproj [hunk 1]
```diff
@@ -8,15 +8,16 @@
     <AssemblyName>Microsoft.Azure.Functions.Worker</AssemblyName>
     <RootNamespace>Microsoft.Azure.Functions.Worker</RootNamespace>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
-    <MinorProductVersion>8</MinorProductVersion>    
+    <MinorProductVersion>8</MinorProductVersion>
+    <PatchProductVersion>1</PatchProductVersion>
```
Comment (kshyju) [inline src/DotNetWorker/DotNetWorker.csproj:12]: I only bumped the patch version as the upgrade in dependencies is not bringing any public API changes/behavior changes to customers.  Open to discuss the SEMVERsioning of this change.

Comment (kshyju) [reply src/DotNetWorker/DotNetWorker.csproj:12]: Will update release notes after we collectively agree on the right package version bump to be used.

**Review by jviau — APPROVED** [review]
Have we looked at using central packaging versioning?

https://devblogs.microsoft.com/nuget/introducing-central-package-management/

**Review by satvu — APPROVED** [review]

**Comment by kshyju** [comment]
We will include this dependency updates in the next major version change of worker package (2.X). Closing.

**Pull request closed without merging**